### PR TITLE
Refactor storage fileshare sample to use terraform

### PR DESCRIPTION
**Pull request by saragluna**
## Purpose

The storage file share sample was not using Terraform to provision resources, this PR will add Terraform support for this sample.
It also refactors the sample code to reflect a real-world usage scenario. 

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## How to Test
*  Get the code

```shell
# git clone git@github.com:saragluna/azure-spring-boot-samples.git 
# or 
# git remote add some-name-for-this-remote  git@github.com:saragluna/azure-spring-boot-samples.git 
cd azure-spring-boot-samples
git checkout xiada/refactor-storage-samples
cd storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample
```

* Test the code

Follow the `README.md`.

## What to Check
Verify that the following are valid
Follow the `README.md`.

### CODE REVIEW

File: storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+## Verify This Sample
+1. Write and read a file.  
+    1.1 Send a POST request to update file contents.
+    ```shell
+    curl http://localhost:8080/file/file1.txt -d "new message" -H "Content-Type: text/plain"
+    ```
+    1.2 Verify by sending a GET request.  
+    ```shell
+    curl -XGET http://localhost:8080/file/file1.txt
+    ```
```
Comment (backwind1233) [inline storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/README.md:157]: I got warnnings like below
image

How about change `curl -XGET http://localhost:8080/file/file1.txt` to 
`curl -XGET http://localhost:8080/file/file1.txt --output file1.txt`

File: storage/spring-cloud-azure-starter-storage-blob/storage-blob-sample/src/main/java/com/azure/spring/sample/storage/resource/SampleDataInitializer.java [hunk 2]
```diff
@@ -13,33 +16,31 @@
 
 @Component
 public class SampleDataInitializer implements CommandLineRunner {
-    final static Logger logger = LoggerFactory.getLogger(SampleDataInitializer.class);
-
+    static final String BLOB_RESOURCE_PATTERN = "azure-blob://%s/%s";
+    private final static Logger logger = LoggerFactory.getLogger(SampleDataInitializer.class);
```
Comment (backwind1233) [inline storage/spring-cloud-azure-starter-storage-blob/storage-blob-sample/src/main/java/com/azure/spring/sample/storage/resource/SampleDataInitializer.java:20]: Please help refactor to 'private final static'? sinece most of our samples use this pattern.

**Comment by backwind1233** [comment]
@saragluna I just renamed the default application name within storage-queue-client, check it [here](https://github.com/Azure-Samples/azure-spring-boot-samples/pull/441/commits/051d37194d96110869d68ebdb8c5563db154d725).

**Review by backwind1233 — APPROVED** [review]
LGTM

File: storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/README.md (hunk 1 above)
Comment (saragluna) [reply storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/README.md:157]: We will try to see how to set the `content-type` to the file resource to solve this warning.

**Pull request merged**